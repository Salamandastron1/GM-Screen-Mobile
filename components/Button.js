import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';

const Button = ({ onPress, children }) => (
  <TouchableOpacity onPress={onPress} style={styles.buttonStyle}>
    <Text style={styles.textStyle}>{children}</Text>
  </TouchableOpacity>
);

const styles = {
  buttonStyle: {
    flex: 1,
    backgroundColor: '#54b8c0',
    borderWidth: 1,
    borderRadius: 5,
    borderColor: '#54b8c0',
    alignItems: 'stretch',
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 30,
  },
  textStyle: {
    alignSelf: 'center',
    color: '#f4d8b7',
    fontSize: 30,
    fontWeight: '600',
    paddingTop: 10,
    paddingBottom: 10,
  },
};

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
};

export default Button;
