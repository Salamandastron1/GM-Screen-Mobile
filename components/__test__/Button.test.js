/* eslint-disable */

import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '../Button';

it('should match the shapshot', () => {
  const tree = shallow(<Button />);
  expect(tree).toMatchSnapshot();
});