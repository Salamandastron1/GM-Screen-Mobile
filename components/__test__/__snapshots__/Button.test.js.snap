// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the shapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#54b8c0",
      "borderColor": "#54b8c0",
      "borderRadius": 5,
      "borderWidth": 1,
      "flex": 1,
      "marginLeft": 10,
      "marginRight": 10,
    }
  }
>
  <Component
    style={
      Object {
        "alignSelf": "center",
        "color": "#f4d8b7",
        "fontSize": 30,
        "fontWeight": "600",
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  />
</TouchableOpacity>
`;
